server/src/main/webapp/WEB-INF/rails/app/controllers/admin/jobs_controller.rb

```diff
@@ -83,6 +85,31 @@ def update(jobs)
     end
 
     def update
+      if params[:stage_parent] == 'templates'
+        old_update
+        return
+      end
+      pipeline_name = params[:pipeline_name]
+      original_pipeline_config = pipeline_config_service.getPipelineConfig(pipeline_name)
+      @pipeline = CLONER.deep_clone(original_pipeline_config)
+      @stage = @pipeline.getStage(params[:stage_name])
+      @job = @stage.jobConfigByConfigName(params[:job_name])
+      @job.setConfigAttributes(params[:job])
+      @job.artifactTypeConfigs().getPluggableArtifactConfigs().each do |external_artifact_config|
+        external_artifacts_service.validateExternalArtifactConfig(external_artifact_config, go_config_service.artifactStores().find(external_artifact_config.getStoreId), false)
```

Comment:
Can you check if this validation is required, since the artifacts are validated in the Command - https://github.com/gocd/gocd/blob/master/server/src/main/java/com/thoughtworks/go/config/update/UpdatePipelineConfigCommand.java#L71